src/Api/Rest/Rest.php

```diff
@@ -0,0 +1,225 @@
+				break;
+
+			case 'post':
+				$operation = !empty($task) ? 'task' : 'create';
+				break;
+
+			case 'patch':
+				$operation = 'patch';
+				break;
+
+			case 'delete':
+				$operation = 'delete';
+				break;
+		}
+
+		$this->setOption('operation', $operation);
+
+		// If task is pointing to some other operation like apply, update or delete.
+//		if (!empty($task) && !empty($this->configuration->operations->task->{$task}['useOperation']))
+//		{
+//			$useOperation = strtoupper((string) $this->configuration->operations->task->{$task}['useOperation']);
+//
+//			if (in_array($useOperation, array('create', 'read', 'update', 'delete', 'documentation')))
+//			{
+//				$operation = $useOperation;
+//			}
+//		}
+
+		// Get the Profile object for the webservice requested.
+		$this->profile = Factory::getProfile($this->getContainer(), $clientName, $resourceName, $version, $operation);
+
+		$this->webservice = Factory::getWebservice($this->getContainer(), $operation, $this->getOptions());
+
+		// Set initial status code to OK.
+		$this->setStatusCode(200);
+
+		// We do not want some unwanted text to appear before output.
+		ob_start();
+
+		try
+		{
+			// Execute the web service operation.
+			$this->resource = $this->webservice->execute($this->profile);
+
+			$executionErrors = ob_get_contents();
+			ob_end_clean();
+//			ob_end_flush();		// TEMPORARY so we can see any error messages.
+		}
+		catch (\Exception $e)
+		{
+			$executionErrors = ob_get_contents();
+			ob_end_clean();
+
+			throw $e;
+		}
+
+		$messages = $this->app->getMessageQueue();
+
+		if (!empty($executionErrors))
+		{
+			$messages[] = array('message' => $executionErrors, 'type' => 'notice');
+		}
+
+		if (!empty($messages))
+		{
+			// If we are not in debug mode we will take out everything except errors.
+			if ($this->app->get('webservices.debug_webservices', 0) == 0)
+			{
+				foreach ($messages as $key => $message)
+				{
+					if ($message['type'] != 'error')
+					{
+						unset($messages[$key]);
+					}
+				}
+			}
+
+			$this->resource->setData('_messages', $messages);
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Method to send the application response to the client.
+	 * All headers will be sent prior to the main application output data.
+	 *
+	 * @return  void
+	 *
+	 * @since   1.2
+	 */
+	public function render()
+	{
+		// Set token to uri if used in that way
+		$token = $this->getOptions()->get('accessToken', '');
+		$client = $this->getOptions()->get('webserviceClient', '');
+
+		if (!empty($token))
+		{
+			$this->webservice->setUriParams($this->app->get('webservices.oauth2_token_param_name', 'access_token'), $token);
+		}
+
+		if ($client == 'administrator')
+		{
+			$this->webservice->setUriParams('webserviceClient', $client);
+		}
+
+		// Calculate runtime to this point.
+		$runtime = microtime(true) - $this->app->startTime;
+
+		// Set headers.
+		$this->app->setHeader('Status', $this->webservice->statusCode . ' ' . $this->webservice->statusText, true);
```

Comment:
Yes, it doesn't feel right to me either.  That's why it's already on my list of things I need to look at.